src/envoy/utils/mixer_control.h

```diff
@@ -42,5 +48,40 @@ Grpc::AsyncClientFactoryPtr GrpcClientFactoryForCluster(
     const std::string &cluster_name, Upstream::ClusterManager &cm,
     Stats::Scope &scope);
 
+std::unique_ptr<const LocalAttributes> CreateLocalAttributes(
+    const LocalNode &local);
+
+bool Extract(const envoy::api::v2::core::Node &node, LocalNode *args);
+
+inline bool ReadMap(
+    const google::protobuf::Map<std::string, google::protobuf::Value> &meta,
+    const std::string &key, std::string *val) {
+  const auto it = meta.find(key);
+  if (it != meta.end()) {
+    *val = it->second.string_value();
+    return true;
+  }
+  return false;
+}
+
+inline bool ReadMap(
+    const google::protobuf::Map<std::string, Attributes_AttributeValue> &meta,
+    const std::string &key, std::string *val) {
+  const auto it = meta.find(key);
+  if (it != meta.end()) {
+    *val = it->second.string_value();
+    return true;
+  }
+  return false;
+}
+
+// NodeKey are node matadata keys that are expected to be set.
+struct NodeKey {
```

Comment:
This should be a proto that is extracted from metadata for mixer filter (filter ID "mixer").